From b19532dea9f9949d720af2ef2b8799c4d7012072 Mon Sep 17 00:00:00 2001
From: Richard Gong <richard.gong@intel.com>
Date: Mon, 15 Jun 2020 09:29:06 -0500
Subject: [PATCH 107/151] firmware: stratix10-svc: extend svc to support new
 RSU features

commit  ebae460376452dd0d3313d8df15d7a0e210ef77b from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.64-lts

Extend Intel Stratix10 service layer driver to support new RSU DCMF
versions and max retry parameter.

DCMF = Decision Configuration Management Firmware. The max retry parameter
is the maximum times the images is allowed to reload itself before giving
up and starting RSU failover flow.

Signed-off-by: Richard Gong <richard.gong@intel.com>
Link: https://lore.kernel.org/r/1592231348-31334-3-git-send-email-richard.gong@linux.intel.com
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Richard Gong <richard.gong@intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 include/linux/firmware/intel/stratix10-smc.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/linux/firmware/intel/stratix10-smc.h b/include/linux/firmware/intel/stratix10-smc.h
index 5f5528dad6a3..ee9769aefb2c 100644
--- a/include/linux/firmware/intel/stratix10-smc.h
+++ b/include/linux/firmware/intel/stratix10-smc.h
@@ -363,7 +363,8 @@ INTEL_SIP_SMC_FAST_CALL_VAL(INTEL_SIP_SMC_FUNCID_FPGA_CONFIG_COMPLETED_WRITE)
 /**
  * Request INTEL_SIP_SMC_RSU_DCMF_VERSION
  *
- * Sync call used by service driver at EL1 to query DCMF version from FW
+ * Sync call used by service driver at EL1 to query DCMF (Decision
+ * Configuration Management Firmware) version from FW
  *
  * Call register usage:
  * a0 INTEL_SIP_SMC_RSU_DCMF_VERSION
@@ -385,7 +386,7 @@ INTEL_SIP_SMC_FAST_CALL_VAL(INTEL_SIP_SMC_FUNCID_FPGA_CONFIG_COMPLETED_WRITE)
 /**
  * Request INTEL_SIP_SMC_RSU_MAX_RETRY
  *
- * Sync call used by service driver at EL1 to query max retry value from FM
+ * Sync call used by service driver at EL1 to query max retry value from FW
  *
  * Call register usage:
  * a0 INTEL_SIP_SMC_RSU_MAX_RETRY
-- 
2.26.1

